src/msg/simple/Pipe.cc

```diff
@@ -332,7 +334,7 @@ int Pipe::accept()
     goto fail_unlocked;
   }
   {
-    bufferptr tp(sizeof(peer_addr));
+    bufferptr tp(sizeof(ceph_entity_addr));
```

Comment:
i understand this is semanticly more correct.
 
but can we make this explicit? say, we encode/decode this field with a `entity_addr_t`, while read a chunk sizing `sizeof(ceph_entity_addr)`. maybe we can have a 
```c++
static_assert(sizeof(entity_addr_t) == sizeof(ceph_entity_addr))
```
somewhere?